beta/nncf/tensorflow/sparsity/callbacks.py

```diff
@@ -14,20 +14,24 @@
 import tensorflow as tf
 
 from beta.nncf.tensorflow.callbacks.statistics_callback import StatisticsCallback
+from beta.nncf.tensorflow.sparsity.base_algorithm import BaseSparsityController
 from beta.nncf.tensorflow.sparsity.utils import convert_raw_to_printable
 from beta.nncf.tensorflow.sparsity.utils import prepare_for_tensorboard
 
 
 class UpdateMask(tf.keras.callbacks.Callback):
-    def __init__(self, scheduler):
+    def __init__(self, sprase_ctrl: BaseSparsityController):
         super().__init__()
-        self._scheduler = scheduler
+        self._ctrl = sprase_ctrl
 
     def on_train_batch_begin(self, batch, logs=None):
-        self._scheduler.step()
+        self._ctrl.scheduler.step()
 
     def on_epoch_begin(self, epoch, logs=None):
-        self._scheduler.epoch_step(epoch)
+        self._ctrl.scheduler.epoch_step(epoch)
+
+    def on_test_begin(self, epoch, logs=None):
+        self._ctrl.freeze()
```

Comment:
I can assume it breaks the training pipeline because `on_test_begin` is called at the beginning of each evaluation or validation stage and the `freeze` method too.

Reply:
Miss it, thanks